proposals/readonly-setter-calls-on-non-variables.md

```diff
@@ -145,3 +165,33 @@ class C
     public Span<object> StructWithRefReturningIndexer => ArraySegmentProp.AsSpan();
 }
 ```
+
+## Downsides
+
+If this proposal is taken, it becomes a source-breaking change to remove the `readonly` keyword from a struct or setter. Without the `readonly` keyword, the errors would then be relevant and would reappear.
+
+However, in effect, this bridge has already been crossed for invocation expressions. It's already a source-breaking change to remove the `readonly` keyword from a struct or setter, because the struct value may be obtained from an invocation expression where the compiler already omits the CS1612 error for these:
+
+```cs
+// Removing 'readonly' from S1 causes a CS1612 error.
+M().Prop = 1;
```

Comment:
The following is disallowed, which tripped me up a lot as I was thinking through this space:

```cs
S1? s1 = new S1();
s1.Value.Prop = 1; // error CS1612: Cannot modify the return value of 'S1?.Value' because it is not a variable
```

So, it sounds like a big part of the motivation here is, the current state is super inconsistent and weird.

Reply:
> The following is disallowed

@RikkiGibson I can see the consistency in this, if we imagine `Nullable<T>.Value` is treated as any other struct-returning property for this example.